src/js/controller/vendingMachine.js

```diff
@@ -0,0 +1,131 @@
+  #handlers;
+
+  constructor() {
+    this.currentMenu = getLocalStorage("menu") || "manager";
+
+    const $menu = $("#menu");
+    const $menuItem = $(`#menu button[name=${this.currentMenu}]`);
+    const managerModel = new ProductManagerModel();
+    const chargerModel = new ChangeChargerModel();
+    const purchaseModel = new ProductPurchaseModel();
+
+    this.#models = {
+      manager: managerModel,
+      charger: chargerModel,
+      purchase: purchaseModel,
+    };
+    this.#currentModel = this.#models[this.currentMenu];
+    this.#handlers = {
+      manager: () => {
+        const $form = $("#product-manager-form");
+        $form.addEventListener("submit", this.submitManagerForm.bind(this));
+      },
+      charger: () => {
+        const $form = $("#coin-charging-form");
+        $form.addEventListener("submit", this.submitChargerForm.bind(this));
+      },
+      purchase: () => {},
+    };
+
+    this.#currentModel.initialize();
+    this.#handlers[this.currentMenu]();
+
+    $menuItem.classList.add("active");
+    $menu.addEventListener("click", this.menuHandler.bind(this));
+  }
+
+  submitChargerForm(e) {
+    e.preventDefault();
+    const chargingInput = $(".charger-input");
+    try {
+      validateChargerInput(chargingInput.value);
+      this.#currentModel.setState("totalAmount", Number(chargingInput.value));
+    } catch (err) {
+      alert(err.message);
+      chargingInput.focus();
+    }
+  }
+
+  submitManagerForm(e) {
+    e.preventDefault();
+    let currentInput;
+    try {
+      const $$inputs = $$(".product-input");
+      const newState = {};
+      $$inputs.forEach((input) => {
+        currentInput = input;
+
+        const { name, value } = currentInput;
+
+        validateManagerInputs[name](value);
+
+        newState[input.name] = input.value;
+      });
+
+      this.#currentModel.setState("products", newState);
+    } catch (err) {
+      alert(err.message);
+      currentInput.focus();
+    }
+  }
+
+  menuHandler({ target }) {
+    const isMenuButton = target.classList.contains("button");
+
+    if (isMenuButton === false) return;
+
+    try {
+      this.validateMenu(target);
+      this.changeMenu(target);
+      this.changeView(target);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  validateMenu($target) {
+    const isMenu = Object.keys(this.#models).includes($target.name);
+
+    if (isMenu === false) {
+      alert(ERROR_MESSAGE.INVALID_MENU);
+      throw new ValidationError(ERROR_MESSAGE.INVALID_MENU);
+    }
+  }
+
+  changeMenu($target) {
+    const $$buttons = $$("#menu button");
+
+    $$buttons.forEach((button) => {
+      button.classList.remove("active");
+    });
+
+    this.currentMenu = $target.name;
+    this.#currentModel = this.#models[$target.name];
+
+    $target.classList.add("active");
+    setLocalStorage("menu", this.currentMenu);
+  }
+
+  changeView($target) {
+    this.#models[$target.name].initialize();
+    this.#handlers[$target.name]();
+  }
```

Comment:
```suggestion
  changeView(menuName) {
    this.#models[menuName].initialize();
    this.#handlers[meneName]();
  }
```

changeView라는 이름도 조금 안어울리는 것 같네요
함수 내부 로직을 충분히 설명하고 있는지 고민이 필요해 보입니다.

Reply:
최대한 내부 로직을 표현할 수 있도록 네이밍을 고민해봤는데... 좀 많이 기네요.. ㅠㅠ 이렇게 길어도 괜찮을까요..?
오히려 가독성을 헤치지 않을까요??
changeView > initializeBasedOnChangedMenu

추가적으로 validateMenu 내부로직에 변수명도 수정했습니다!

리뷰 반영 커밋
https://github.com/next-step/js-vending-machine/pull/61/commits/80fd4fe9b9a5174d5c459a27d7ccf6215acc83a2, https://github.com/next-step/js-vending-machine/pull/61/commits/b8fdddce56d488f07b1fc74fa6f8103bc7e8e936